src/overlays/actors/ovl_En_Pst/z_en_pst.c

```diff
@@ -31,51 +108,311 @@ const ActorInit En_Pst_InitVars = {
+                return D_80B2C320;
+            case 4:
+                return D_80B2C36C;
+            default:
+                return NULL;
+        }
+    }
+}
+
+s32 EnPst_CheckTalk(EnPst* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s32 ret = false;
+
+    if (this->stateFlags & 7) {
+        if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+            this->stateFlags &= ~0x30;
+            if (player->exchangeItemId == EXCH_ITEM_LETTER_TO_KAFEI) {
+                this->stateFlags |= 0x10;
+                this->exchangeItemId = player->exchangeItemId;
+            } else if (player->exchangeItemId != EXCH_ITEM_NONE) {
+                this->stateFlags |= 0x20;
+                this->exchangeItemId = player->exchangeItemId;
+            }
+            this->unk21C = func_80B2B874(this);
+            SubS_UpdateFlags(&this->stateFlags, 0, 7);
+            this->behaviour = 0;
+            this->msgEventCallback = NULL;
+            this->stateFlags |= 0x40;
+            this->msgEventScript = EnPst_GetMsgEventScript(this, play);
+            this->actionFunc = func_80B2BE54;
+            ret = true;
+        }
+    }
+    return ret;
+}
+
+s32 EnPst_UpdateFlagsSubs(EnPst* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    SubS_UpdateFlags(&this->stateFlags, 3, 7);
+    return true;
+}
 
-#endif
+s32 func_80B2BD30(EnPst* this, PlayState* play, ScheduleOutput* scheduleOutput) {
+    s32 ret = false;
 
-extern ColliderCylinderInit D_80B2C4B8;
-extern CollisionCheckInfoInit2 D_80B2C4E4;
+    this->stateFlags = 0;
 
-extern UNK_TYPE D_06001A80;
+    switch (scheduleOutput->result) {
+        case POSTBOX_SCH_AVAILABLE:
+            ret = EnPst_UpdateFlagsSubs(this, play, scheduleOutput);
+            break;
+        case POSTBOX_SCH_CHECKED_BY_MAILMAN:
+            ret = true;
+            break;
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B830.s")
+void EnPst_Idle(EnPst* this, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B874.s")
+void EnPst_FollowSchedule(EnPst* this, PlayState* play) {
+    static u8* sScheduleScripts[] = {
+        D_80B2C200, D_80B2C20C, D_80B2C218, D_80B2C224, D_80B2C230,
+    };
+    s16 params = this->actor.params;
+    ScheduleOutput scheduleOutput;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Pst/func_80B2B8F4.s")
+    if (!Schedule_RunScript(play, sScheduleScripts[params], &scheduleOutput) ||
+        ((this->scheduleResult != scheduleOutput.result) && !func_80B2BD30(this, play, &scheduleOutput))) {
+        this->actor.shape.shadowDraw = NULL;
+        this->actor.flags &= ~ACTOR_FLAG_1;
+        scheduleOutput.result = POSTBOX_SCH_NONE;
+    } else {
+        this->actor.shape.shadowDraw = ActorShadow_DrawCircle;
+        this->actor.flags |= ACTOR_FLAG_1;
+    }
+    this->scheduleResult = scheduleOutput.result;
+    EnPst_Idle(this, play);
```

Comment:
```suggestion
    EnPst_HandleSchedule(this, play);
```
Small note, but this isn't an action function so we would have probably wanted to avoid using an action word to describe it anyways.